app/js/controllers/copy_move.js

```diff
@@ -26,9 +26,13 @@ export default class MoveController extends Controller {
     this.view.dialog.open();
   }
 
+  setMode(mode) {
+    this.mode = mode;
+  }
+
   before() {
     AddonService.generate(this.target, (generator) => {
-      generator.opMoveBefore(this.destination);
+      generator[`op${this.mode}Before`](this.destination);
```

Comment:
This looks a bit like black magic on first glance. Might be better to just use a pattern that's a bit more ubiquitous, like:
```js
var operation = generator['op' + this.mode + 'Before'];
operation(this.destination);
```